### PR TITLE
BigFullChain, updates for FWD detectors and Tracking

### PR DESCRIPTION
**Pull request by jdbrice**
Adds a few BFC options for FWD chains.
Also, makes FTT options use consistent case.

### CODE REVIEW

File: StRoot/StBFChain/BigFullChain.h [hunk 1]
```diff
@@ -1734,6 +1737,10 @@ Bfc_st BFC[] = { // standard chains
                                                                              "sTGC Point maker",    kFALSE},
   {"FttQA","","fttChain","","StFttQAMaker","StFttQAMaker", "sTGC Raw hit QA maker",                 kFALSE},
 
+  // forward track maker
+  {"FwdTrack","","","","StFwdTrackMaker",
+  "XMLIO.so,genfit2.so,KiTrack.so,StarGeneratorUtil.so,libMathMore.so,StEventUtilities.so,StEpdUtil,StFwdTrackMaker",
```
Comment (genevb) [inline StRoot/StBFChain/BigFullChain.h]: I know there's a few other instances of including the ".so" suffix in libraries to load, but if it works without them, I'd recommend removing the suffix for a cleaner look (which is part of the point of this PR).

Comment (genevb) [inline StRoot/StBFChain/BigFullChain.h]: Some of these libraries haven't been used by anyone else. I'm guessing you've tested that including them works, but can you please confirm or comment on this?

**Review by genevb — COMMENTED** [review]
Otherwise this looks fine.

**Comment by plexoos** [comment]
I would like to repeat a request from three months ago https://github.com/star-bnl/star-sw/issues/359#issuecomment-1155713790
Just point me to some input files which can be used for tests and please provide the appropriate running options

**Comment by perevbnlgov** [comment]
I do not know how speed is important there. If it is important then it could
be optimized. In proposed code:
          StThreeVectorD xyz=mFcsDb->getStarXYZfromColumnRow(det,clu->x(),clu->y());
	  double dx = xyz.x() - proj.x();
	  double dy = xyz.y() - proj.y();
	  double dr = sqrt(dx*dx + dy*dy);
//I removed debug here
	  if(dr<mMaxDistance[ehp]){
	    if(ehp==0){trk->addEcalCluster(clu);}
	    if(ehp==1){trk->addHcalCluster(clu);}
	    clu->addTrack(trk);
	    nMatch[ehp]++;
	  }
But the following will be faster because for the wrong case dx or dy is already too big

	  double maxDr = mMaxDistance[ehp])
	  double dx = fabs(xyz.x() - proj.x()); if (dx)> maxDr) break;
	  double dy = fabs(xyz.y() - proj.y()); if (dy > maxDr) break;
	  double dr = sqrt(dx*dx + dy*dy);     if (dr > maxDr) break;

	  if(ehp==0){trk->addEcalCluster(clu);} else {trk->addHcalCluster(clu);}
	  clu->addTrack(trk);
	  nMatch[ehp]++;

**Comment by jdbrice** [comment]
> I do not know how speed is important there. If it is important then it could be optimized. In proposed code: StThreeVectorD xyz=mFcsDb->getStarXYZfromColumnRow(det,clu->x(),clu->y()); double dx = xyz.x() - proj.x(); double dy = xyz.y() - proj.y(); double dr = sqrt(dx_dx + dy_dy); //I removed debug here if(dr<mMaxDistance[ehp]){ if(ehp==0){trk->addEcalCluster(clu);} if(ehp==1){trk->addHcalCluster(clu);} clu->addTrack(trk); nMatch[ehp]++; } But the following will be faster because for the wrong case dx or dy is already too big
> 
> ```
>   double maxDr = mMaxDistance[ehp])
>   double dx = fabs(xyz.x() - proj.x()); if (dx)> maxDr) break;
>   double dy = fabs(xyz.y() - proj.y()); if (dy > maxDr) break;
>   double dr = sqrt(dx*dx + dy*dy);     if (dr > maxDr) break;
> 
>   if(ehp==0){trk->addEcalCluster(clu);} else {trk->addHcalCluster(clu);}
>   clu->addTrack(trk);
>   nMatch[ehp]++;
> ```

@perevbnlgov what code does this comment refer to? This PR only changes a few lines in BigFullChain.h

**Comment by plexoos** [comment]
> what code does this comment refer to?

Indeed, the comment does not seem to belong here. Looks like a duplicate of the [original comment ](https://github.com/star-bnl/star-sw/pull/398#issuecomment-1254391294) in #398

**Comment by jdbrice** [comment]
> I would like to repeat a request from three months ago [#359 (comment)](https://github.com/star-bnl/star-sw/issues/359#issuecomment-1155713790)
> Just point me to some input files which can be used for tests and please provide the appropriate running options

A selection of good DAQ files are now restored to here:
/gpfs01/star/pwg_tasks/FwdCalib/DAQ/zfa/

any 10 (or so) events can be taken from one of those files for tests.

File: StRoot/StBFChain/BigFullChain.h [hunk 2]
```diff
@@ -1734,6 +1737,10 @@ Bfc_st BFC[] = { // standard chains
                                                                              "sTGC Point maker",    kFALSE},
   {"FttQA","","fttChain","","StFttQAMaker","StFttQAMaker", "sTGC Raw hit QA maker",                 kFALSE},
 
+  // forward track maker
+  {"FwdTrack","","","","StFwdTrackMaker",
+  "XMLIO,genfit2,KiTrack,StarGeneratorUtil,libMathMore,StEventUtilities,StEpdUtil,StFwdTrackMaker", 
+                                                            "Forward Track Maker",                  kFALSE},
```
Comment (plexoos) [inline StRoot/StBFChain/BigFullChain.h:1743]: I may be missing something but I don't see where StFwdTrackMaker depends on either libMathMore or StEpdUtil libraries. This needs to be confirmed

Also, what is that XMLIO library? I think the library name is libxml2

**Comment by jdbrice** [comment]
This was addressed in ci: Add integration test for forward tracking maker #410

**Pull request closed without merging**